File: doc-examples/jdbc-example-java/src/main/java/example/Student.java
package example;

import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;

@MappedEntity(value = "TBL_STUDENT", alias = "s")
public class Student {
    @Id
    @GeneratedValue(GeneratedValue.Type.IDENTITY)
    private Long id;

    private String name;
}

File: doc-examples/jdbc-example-python/src/main/python/example/Student.py
from dataclasses import dataclass
from typing import Annotated

from micronaut.data.annotation import GeneratedValue, Id, MappedEntity


@MappedEntity(value="TBL_STUDENT", alias="s")
@dataclass
class Student:
    name: str
    id: Annotated[int | None, Id, GeneratedValue("IDENTITY")] = None
